From: itpplasma
Subject: Fix GCC version comparison for versions >= 10

The shell comparison `[ $(GFORTRAN_VERSION) -gt 9 ]` fails when
gfortran -dumpversion returns a semver string like "15.2.1" because
the -gt operator expects integers. This extracts the major version
number before comparison.

Upstream PR: https://github.com/PrincetonUniversity/STELLOPT/pull/417

diff --git a/SHARE/make_debian.inc b/SHARE/make_debian.inc
index 67dd28c1..b99fb7dd 100644
--- a/SHARE/make_debian.inc
+++ b/SHARE/make_debian.inc
@@ -33,7 +33,8 @@ SILOHOME = /usr/lib/$(ARCH)-linux-gnu
   LIBS    = -L/usr/lib -lopenblas -lscalapack-openmpi

   GFORTRAN_VERSION := $(shell $(FC) -dumpversion)
-  ifeq "$(shell [ $(GFORTRAN_VERSION) -gt 9 ] && echo true)" "true"
+  GFORTRAN_VERSION_MAJOR := $(firstword $(subst ., ,$(GFORTRAN_VERSION)))
+  ifeq "$(shell [ $(GFORTRAN_VERSION_MAJOR) -gt 9 ] && echo true)" "true"
       FLAGS_R += -fallow-argument-mismatch
       FLAGS_D += -fallow-argument-mismatch
   endif
diff --git a/SHARE/make_redhat.inc b/SHARE/make_redhat.inc
index 037f42ce..84d394ff 100644
--- a/SHARE/make_redhat.inc
+++ b/SHARE/make_redhat.inc
@@ -21,7 +21,8 @@
             /lib64/openmpi/lib/libscalapack.so.2

   GFORTRAN_VERSION := $(shell $(FC) -dumpversion)
-  ifeq "$(shell [ $(GFORTRAN_VERSION) -gt 9 ] && echo true)" "true"
+  GFORTRAN_VERSION_MAJOR := $(firstword $(subst ., ,$(GFORTRAN_VERSION)))
+  ifeq "$(shell [ $(GFORTRAN_VERSION_MAJOR) -gt 9 ] && echo true)" "true"
       FLAGS_R += -fallow-argument-mismatch
       FLAGS_D += -fallow-argument-mismatch
   endif
diff --git a/SHARE/make_ubuntu.inc b/SHARE/make_ubuntu.inc
index 188919f6..81becbd2 100644
--- a/SHARE/make_ubuntu.inc
+++ b/SHARE/make_ubuntu.inc
@@ -31,7 +31,8 @@

   # Need to allow argument mismatch for gfortran version 10 and above
   GFORTRAN_VERSION := $(shell $(FC) -dumpversion)
-  ifeq "$(shell [ $(GFORTRAN_VERSION) -gt 9 ] && echo true)" "true"
+  GFORTRAN_VERSION_MAJOR := $(firstword $(subst ., ,$(GFORTRAN_VERSION)))
+  ifeq "$(shell [ $(GFORTRAN_VERSION_MAJOR) -gt 9 ] && echo true)" "true"
       FLAGS_R += -fallow-argument-mismatch
       FLAGS_D += -fallow-argument-mismatch
   endif
